### DISCUSSION TITLE
Scheduler keep schedule task deleted DAGs

### DISCUSSION THREAD

**Discussion by fahmiduldul** [apache/airflow#32398] [Q&A]
Hi everyone, want ask question regarding scheduler.

So I created DAG called `dag_id_1` in my DAG bag and after several days I deleted the DAG from DAG bag. After deleted, the scheduler still schedule task of the deleted DAG.

```
task_instance table
| task_id   | dag_id   | queued_by_job_id |
|-----------|----------|------------------|
| task_id_1 | dag_id_1 | 6445500          |
| task_id_1 | dag_id_1 | 6445500          |


job table
   id    | dag_id |  state  |   job_type   |       latest_heartbeat        |   executor_class   |
---------+--------+---------+--------------+-------------------------------+--------------------|
 6445500 |        | running | SchedulerJob | 2023-07-05 10:01:22.940214+00 | KubernetesExecutor |
```

I also checked the `dag` table and the corresponding DAG has been deactivated.

This issue caused the externally killed task soar and keep taking our execution slot. Is there a way to fix this? Thanks!

**Comment by andresitorusz** [comment]
Why don't you just pause the unused DAG?

**Reply by fahmiduldul** [reply]
I cant pause the DAG because the DAG is already deleted. It doesn't show up in UI anymore but somehow scheduler keeps schedule it.

**Reply by andresitorusz** [reply]
I see, but you can still pause the DAG using CLI.

Ref: https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#pause

**Reply by andresitorusz** [reply]
Or maybe you can try to:
1. Stop the scheduler
2. Delete row related to the dag_id from Airflow metadata
3. Restart the scheduler
